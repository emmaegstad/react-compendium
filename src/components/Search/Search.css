.Search {
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 5px;
  margin-bottom: 4vh;
}

.search-input {
  border: none;
  /* background-color: gold; */
  border: 1px orangered solid;
  border-radius: 5px;
  height: 2vh;
}

.search-submit {
  border: none;
  background: none;
}

.search-submit img {
  height: 3vh;
}

.search-submit:hover {
  transform: scale(1.1);
}

.search-submit:active {
  filter: drop-shadow(1px 1px 2px black);
}
