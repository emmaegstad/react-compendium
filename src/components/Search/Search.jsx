import React from 'react';
import './Search.css';

export default function Search({
  query,
  setQuery,
  setLoading,
  types,
  selectedType,
  setSelectedType,
}) {
  return (
    <div className="Search">
      <input
        className="search-input"
        type="text"
        placeholder="Find Pokemon"
        value={query}
        onChange={(e) => {
          setQuery(e.target.value);
        }}
      />
      <select
        className="search-select"
        value={selectedType}
        onChange={(e) => setSelectedType(e.target.value)}
      >
        <option value="all">All</option>
        {types.map((type) => {
          return (
            <option key={type} value={type}>
              {type}
            </option>
          );
        })}
      </select>
      <button className="search-submit" onClick={() => setLoading(true)}>
        <img src={process.env.PUBLIC_URL + '/assets/pokeball.png'} />
      </button>
    </div>
  );
}
