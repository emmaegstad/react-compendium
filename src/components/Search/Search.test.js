import { render, screen } from '@testing-library/react';
import Search from './search';

test('renders the Search component', () => {
  const container = render(
    <Search
      query="char"
      setQuery="char"
      setLoading="false"
      types={['fairy', 'dragon']}
      selectType="dragon"
      setSelectType="dragon"
      order="asc"
      setOrder="asc"
    />
  );
  expect(container).toMatchSnapshot();
});
