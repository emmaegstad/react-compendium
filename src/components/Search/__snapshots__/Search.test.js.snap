// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Search component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Search"
      >
        <input
          class="search-input"
          placeholder="Find Pokemon"
          type="text"
          value="char"
        />
        <select
          class="search-select"
        >
          <option
            value="all"
          >
            All
          </option>
          <option
            value="fairy"
          >
            fairy
          </option>
          <option
            value="dragon"
          >
            dragon
          </option>
        </select>
        <select
          class="search-select"
        >
          <option
            value="asc"
          >
            Ascending
          </option>
          <option
            value="desc"
          >
            Descending
          </option>
        </select>
        <button
          class="search-submit"
        >
          <img
            src="/assets/pokeball.png"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Search"
    >
      <input
        class="search-input"
        placeholder="Find Pokemon"
        type="text"
        value="char"
      />
      <select
        class="search-select"
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="fairy"
        >
          fairy
        </option>
        <option
          value="dragon"
        >
          dragon
        </option>
      </select>
      <select
        class="search-select"
      >
        <option
          value="asc"
        >
          Ascending
        </option>
        <option
          value="desc"
        >
          Descending
        </option>
      </select>
      <button
        class="search-submit"
      >
        <img
          src="/assets/pokeball.png"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
